probe-rs/src/debug/registers.rs

```diff
@@ -1,162 +1,243 @@
-use crate::core::Core;
-
-use probe_rs_target::Architecture;
-
-use std::collections::HashMap;
-
-use crate::core::{RegisterDescription, RegisterFile};
+use crate::{
+    core::{Core, RegisterDataType, RegisterFile},
+    Error, RegisterId, RegisterValue,
+};
+
+/// The group name of a register.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum RegisterGroup {
+    /// Core / CPU Registers
+    Base,
+    /// Argument Register
+    Argument,
+    /// Result Register
+    Result,
+    /// [`RegisterFile`] contains some register descriptions that are not part of an array, and may or may not have the same `RegisterId` as registers in other groups.
+    Singleton,
+}
 
-/// All the register information currently available.
+/// Stores the relevant information from [`RegisterDescription`] as well as additional information required during debug.
 #[derive(Debug, Clone, PartialEq)]
-pub struct Registers {
-    pub(crate) register_description: &'static RegisterFile,
+pub struct DebugRegister {
+    /// To lookup platform specific details of register definitions.
+    pub register_file: &'static RegisterFile,
+    /// Register definitions are grouped depending on their purpose.
+    pub group: RegisterGroup,
+    /// The name of the register.
+    pub name: &'static str,
+    /// If a special name exists for an existing register, e.g. Arm register 'r15' is also known as 'pc' (program counter)
+    pub special_name: Option<&'static str>,
+    /// The location where the register is stored.
+    pub id: RegisterId,
+    /// [DWARF](https://dwarfstd.org) specification, section 2.6.1.1.3.1 "... operations encode the names of up to 32 registers, numbered from 0 through 31, inclusive ..."
+    pub dwarf_id: Option<u16>,
```

Comment:
I'd suggest we consider adding `dwarf_id` to `RegisterDescription` directly.  This seems like generally useful information for consumers of the API and it's static configuration data, not runtime state.

Reply:
Agreed :)